api/server/router/local/image.go

```diff
@@ -184,12 +177,7 @@ func (s *router) postImagesPush(ctx context.Context, w http.ResponseWriter, r *h
 
 	name := vars["name"]
 	output := ioutils.NewWriteFlusher(w)
-	imagePushConfig := &graph.ImagePushConfig{
-		MetaHeaders: metaHeaders,
-		AuthConfig:  authConfig,
-		Tag:         r.Form.Get("tag"),
-		OutStream:   output,
-	}
+	imagePushConfig := graph.NewImagePushConfig(metaHeaders, authConfig, r.Form.Get("tag"), output)
```

Comment:
Any time the constructor, that change to be done everywhere. Any time the config struct changes, that only has to be done in one place if it's adding a new field. If it's deleting or renaming, it also has to be done everywhere.
I don't mind the config structs, but they should probably all be moved to one place, if it's a 'config' package, then the name becomes nice, 'config.ImagePush' or some such, in this example.